app/src/lib/shells/win32.ts

```diff
@@ -215,6 +228,30 @@ async function findGitBash(): Promise<string | null> {
   return null
 }
 
+async function findCygwin(): Promise<string | null> {
+  const registryPath = enumerateValues(
+    HKEY.HKEY_LOCAL_MACHINE,
+    'SOFTWARE\\Cygwin\\setup'
```

Comment:
Do you think we should check for 32-bit Cygwin installation as well?

Reply:
I forgot about that. I will check for 32-bit Cygwin later and post an update.

Reply:
Were you looking in the right location, i.e. `HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Cygwin\setup`? That's where I found it after a test install yesterday.

I'd rather not hard-code paths for installers where the user can change the location easily, because the registry is the expected way for us to identify where a program is located on disk.